lib/worker.js

```diff
@@ -223,41 +223,37 @@ FiveBeansWorker.prototype.doWork = function doWork(handler, jobID, jobdata)
 		return handler.type;
 	})(handler);
 
-	try
+	handler.work(jobdata, function(action, delay)
 	{
-		handler.work(jobdata, function(action, delay)
-		{
-			var elapsed = new Date().getTime() - start;
+		var elapsed = new Date().getTime() - start;
 
-			self.emit('job.handled', { id: jobID, type: handlerName, elapsed: elapsed, action: action }, jobdata);
+		self.emit('job.handled', { id: jobID, type: handlerName, elapsed: elapsed, action: action }, jobdata);
 
-			switch (action)
-			{
-			case 'success':
-				self.deleteAndMoveOn(jobID);
-				break;
+		switch (action)
+		{
+		case 'success':
+			self.deleteAndMoveOn(jobID);
+			break;
 
-			case 'release':
-				self.releaseAndMoveOn(jobID, delay);
-				break;
+		case 'release':
+			self.releaseAndMoveOn(jobID, delay);
+			break;
 
-			case 'bury':
-				self.buryAndMoveOn(jobID);
-				break;
+		case 'bury':
+			self.buryAndMoveOn(jobID);
+			break;
 
-			default:
-				self.buryAndMoveOn(jobID);
-				break;
-			}
-		});
-	}
-	catch (e)
-	{
-		self.emitWarning({ message: 'exception in job handler', id: jobID, handler: handlerName, error: e });
+		default:
+			self.buryAndMoveOn(jobID);
+			break;
+		}
+	}).catch (function (error) 	{
+		self.emit('error', { message: 'exception in job handler', id: jobID, handler: handlerName, error: e });
```

Comment:
This will be a sort of breaking change. The existing event for sending out errors is like this:

```
self.emit('error', err);
```

It emits just an error object. The existing listeners to the `'error'` event will need to change their implementation in order to be able to process the new payload.

This new approach is better as it gives us more details on what happened but we need to make sure the payload in our `emit.('error', payload)` calls have consistent formats.

One approach would be to make an `emitError` function similar to the existing `emitWarning` function.

Also, we should probably include the jobData in the payload of the error

Reply:
Yupz, I added the jobdata in the newest commit.

Hmmn, yes this would require additional handling on the users of the library, so I was thinking of separating the handling of this event into an event called 'handler-error' that way, others can process this as well.

I'll also follow the style given by the original developer by making this emit two events, one for handler error, one for just 'error', this way it will act similar to 'warning'